deploy/006-deploy-psm.ts

```diff
@@ -1,48 +1,87 @@
   const { deployer } = await getNamedAccounts();
 
-  const networkName = network.name === "bscmainnet" ? "bscmainnet" : "bsctestnet";
-  const addresses = ADDRESSES[networkName];
+  // Ensure oracles are deployed
+  await deployOracle(hre);
+
+  const usdtAddress = (
+    await deploy("USDT", {
+      contract: "MockToken",
+      from: deployer,
+      args: ["Tether", "USDT", 18],
+      log: true,
+      autoMine: true,
+    })
+  ).address;
+  const acmAddress = (await deployments.get("AccessControlManager")).address;
+  const treasuryAddress = (await deployments.get("VTreasury")).address;
+  const oracleAddress = (await deployments.get("ResilientOracle")).address;
+
+  let normalVipTimelockAddress;
+  if (hre.network.name === "hardhat") {
+    normalVipTimelockAddress = (
+      await deploy("NormalTimelock", {
+        contract: "TestTimelockV8",
+        from: deployer,
+        args: [deployer, 600],
+        log: true,
+        autoMine: true,
+      })
+    ).address;
+  } else {
+    // We should be able to fetch the deployed contract from the governance network on live networks
+    normalVipTimelockAddress = (await deployments.get("NormalTimelock")).address;
+  }
+
+  const vaiAddress = (
+    await deploy("VAI", {
+      from: deployer,
+      args: [await getChainId()],
+      log: true,
+      autoMine: true,
+    })
+  ).address;
+
   const FEE_IN = 0;
   const FEE_OUT = 0;
   const VAI_MINT_CAP = parseUnits("5000000", 18); // 5M
 
   await deploy("PegStability_USDT", {
     contract: "PegStability",
     from: deployer,
-    args: [addresses.usdt, addresses.vai],
+    args: [usdtAddress, vaiAddress],
     log: true,
     autoMine: true,
     proxy: {
-      owner: addresses.normalVipTimelock,
+      owner: network.name === "hardhat" ? deployer : normalVipTimelockAddress,
       proxyContract: "OpenZeppelinTransparentProxy",
       execute: {
         methodName: "initialize",
-        args: [addresses.acm, addresses.treasury, addresses.oracle, FEE_IN, FEE_OUT, VAI_MINT_CAP],
+        args: [acmAddress, treasuryAddress, oracleAddress, FEE_IN, FEE_OUT, VAI_MINT_CAP],
       },
     },
   });
 
   const psm = await ethers.getContract("PegStability_USDT");
 
   if (network.name !== "hardhat") {
-    const timelockAddress = addresses.normalVipTimelock;
+    const timelockAddress = normalVipTimelockAddress;
     await psm.transferOwnership(timelockAddress);
     console.log(`PSM Contract (${psm.address}) owner changed from ${deployer} to ${timelockAddress}`);
   }
-  return hre.network.live; // when live network, record the script as executed to prevent rexecution
+  return hre.network.live; // when live network, record the script as executed to prevent re-execution
 };
 
 func.id = "psm_initial_deploy"; // id required to prevent re-execution
 func.tags = ["PSM"];
+func.skip = async hre => hre.network.name === "sepolia" || hre.network.name === "opbnbtestnet" || hre.network.name === "opbnbmainnet" || hre.network.name === "ethereum";
```

Comment:
Can we export some functions in utils files to use here? This way we can increase readability?

Reply:
I agree this isn't the prettiest code but I imagine it will get cleaned up as contracts are deployed to new networks. If contracts are only going to be used on bsc networks we could inverse it.